tests/ssl-opt.sh

```diff
@@ -6947,6 +6961,7 @@ run_test    "DTLS cookie: disabled" \
             -S "hello verification requested" \
             -S "SSL - The requested feature is not available"
 
+requires_config_enabled MBEDTLS_ERROR_C
```

Comment:
Note: an alternative would be to change the last condition from `-s "SSL - The requested feature is not available"` to something with the numerical error code instead. I think the current condition is acceptable for this PR but we might want to improve on it in a later PR.

Reply:
Major: I think the next test should be using the same condition. It's probably already skipped on your machine and the CI for IPv6 reasons, but for the sake of machines that have IPv6 enabled, it should have the proper condition as well.

Reply:
I'll add that skip

Reply:
Good point. However I'm not quite convinced. We don't only want the test to pass when the code works as expected, we also want it to fail when it doesn't. If `MBEDTLS_ERROR_C` isn't enabled, then the last assert can never fail, but for the wrong reason. I think we still want the condition on the test - better have a clean skip than passing for possibly the wrong reason. 

(Though I agree that this is now going slightly beyond the scope of the task, but since it's related to another change, and there is a risk of forgetting if we don't do it now, and it's a simple one-line addition, I think it's still best done in this PR.)

Reply:
In that case, the requirement needs to be added for all the DTLS cookie tests.

Reply:
After more careful thought, I came to the conclusion that this final `-S` assert is redundant anyway (probably only kept for symmetry with the case that has an `-s` assert), as we're already asserting that the client exits successfully, so we don't need to assert the absence of a specific error message in addition.

So I withdraw my point and I'm happy with the code as it stands. Thanks for bearing with me as I slowly change my mind :)